cwms-data-api/src/main/java/cwms/cda/data/dao/project/ProjectChildLocationDao.java

```diff
@@ -0,0 +1,218 @@
+            Map<String, List<CwmsId>> locsOfKind = getChildLocationsOfKind(office, projLike, kind);
+            if (locsOfKind != null) {
+                for (Map.Entry<String, List<CwmsId>> entry : locsOfKind.entrySet()) {
+                    String projId = entry.getKey();
+                    List<CwmsId> locs = entry.getValue();
+                    ProjectChildLocations.Builder builder = builderMap.computeIfAbsent(projId, k ->
+                            new ProjectChildLocations.Builder()
+                                    .withProject(new CwmsId.Builder()
+                                            .withOfficeId(office)
+                                            .withName(projId)
+                                            .build()));
+                    switch (kind) {
+                        case EMBANKMENT:
+                            builder.withEmbankments(locs);
+                            break;
+                        case LOCK:
+                            builder.withLocks(locs);
+                            break;
+                        case OUTLET:
+                            builder.withOutlets(locs);
+                            break;
+                        case TURBINE:
+                            builder.withTurbines(locs);
+                            break;
+                        case GATE:
+                            builder.withGates(locs);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+        }
+
+        return builderMap.values().stream()
+                .map(ProjectChildLocations.Builder::build)
+                .collect(Collectors.toList());
+    }
+
+    @Nullable
+    private Map<String, List<CwmsId>> getChildLocationsOfKind(String office, String projRegex, ProjectKind kind) {
+        switch (kind) {
+            case EMBANKMENT:
+                return getEmbankmentChildLocations(office, projRegex);
+            case TURBINE:
+                return getTurbineChildLocations(office, projRegex);
+            case OUTLET:
+                return getOutletChildLocations(office, projRegex);
+            case LOCK:
+                return getLockChildLocations(office, projRegex);
+            case GATE:
+                return getGateChildLocations(office, projRegex);
+            default:
+                return null;
+        }
+
+    }
+
+    private Map<String, List<CwmsId>> getGateChildLocations(String office, @Nullable String projRegex) {
+        Map<String, List<CwmsId>> retval = new LinkedHashMap<>();
+        // AV_GATE is apparently not used.
+        AV_OUTLET view = AV_OUTLET.AV_OUTLET;
+        dsl.selectDistinct(view.OFFICE_ID, view.PROJECT_ID, view.OUTLET_ID)
+                .from(view)
+                .where(view.OFFICE_ID.eq(office)
+                        .and(view.OPENING_UNIT_EN.isNotNull().or(view.OPENING_UNIT_SI.isNotNull()))
```

Comment:
This is the extra bit added for Gates.  I talked to RyanM and looked at existing code and this seemed easier than trying to pick apart the rating descriptor but I'm not a gate expert - is this a valid check to separate gates from outlets?

Reply:
This works for all observed cases that we have seen.